File: plugins/itp-hooks/hooks/tests/pushover-budget-fixtures/go/pushover_title_only_message.go
package main

import (
	"net/http"
	"net/url"
)

func sendBriefNotification(token, user string) error {
	payload := url.Values{
		"token": []string{token},
		"user":  []string{user},
		"title": []string{"Deployment Complete"},
	}

	_, err := http.PostForm("https://api.pushover.net/1/messages.json", payload)
	return err
}

File: plugins/itp-hooks/hooks/tests/pushover-budget-fixtures/python/pushover_title_only_message.py
#!/usr/bin/env python3
import urllib.request
import urllib.parse

token = "app_token_secret"
user_key = "user_key_secret"

# Minimal Pushover send: title only, empty message
params = urllib.parse.urlencode({
    "token": token,
    "user": user_key,
    "title": "Health Check Passed",
    "message": "",  # empty message field
    "priority": -1
})

url = f"https://api.pushover.net/1/messages.json?{params}"
with urllib.request.urlopen(url) as response:
    print(response.read())
